eth/consensus/clique/clique.py

```diff
@@ -0,0 +1,171 @@
+import logging
+import time
+from typing import Sequence, Iterable
+
+from eth.abc import (
+    AtomicDatabaseAPI,
+    BlockHeaderAPI,
+    VirtualMachineAPI,
+)
+from eth.db.chain import ChainDB
+
+from eth_typing import (
+    Address,
+    Hash32,
+)
+from eth_utils import (
+    encode_hex,
+    to_tuple,
+    ValidationError,
+)
+
+from eth.typing import (
+    HeaderParams,
+    VMConfiguration,
+    VMFork,
+)
+from eth.vm.chain_context import ChainContext
+from eth.vm.execution_context import (
+    ExecutionContext,
+)
+
+from .datatypes import (
+    Snapshot,
+)
+from .header_cache import HeaderCache
+from .snapshot_manager import SnapshotManager
+from ._utils import (
+    get_block_signer,
+    is_in_turn,
+    validate_header_integrity,
+)
+
+
+def create_execution_context(header: BlockHeaderAPI,
+                             prev_hashes: Iterable[Hash32],
+                             chain_context: ChainContext,
+                             import_block: BlockHeaderAPI=None) -> ExecutionContext:
+
+    # In Clique consensus, the tx fee goes to the signer
+    fee_recipient = None
+    if import_block is not None:
+        try:
+            fee_recipient = get_block_signer(import_block.header)
```

Comment:
Why can't this be retrieved from `header`?  Is `header` different from `import_block.header`?